lib/lhm/chunker.rb

```diff
@@ -16,7 +16,9 @@ class Chunker
     def initialize(migration, connection = nil, options = {})
       @migration = migration
       @connection = connection
-      @throttler = options[:throttler]
+      if @throttler = options[:throttler]
+        @throttler.connection = @connection if @throttler.respond_to?(:connection=)
```

Comment:
Not loving this for one reason `SlaveLag` is pretty weak now. I mean there will never be a reason to initialize a `SlaveLag` without a connection, this code creates a time dependency that is kinda weird, you create an instance of the `SlaveLag` throttle but if you call any useful method it will just :bomb:, which makes the constructor kinda useless.

So... what about instead of passing instances around we pass the throttler class  as one arg and the throttler construcrtor args as another argument, that we the `SlaveLag` class is not such a weak sauce, but you can still assign w/e connection you want.

On a more functional level are we sharing connections ? I mean is it kosher to set the chunker connection to be the same as the throttle one?